src/core/initializeComponentsFactory.js

```diff
@@ -10,20 +10,23 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import createLifecycle from "./createLifecycle";
-import createComponentRegistry from "./createComponentRegistry";
-import createNetwork from "./network";
 import { stackError } from "../utils";
-import createOptIn from "./createOptIn";
```

Comment:
We were importing a bunch of things here, but I thought the general consensus was that we would only import non-utility modules from `main.js` and injected them all down the tree. As such, I moved the imports to `main.js`. This will allow for easier unit testing of `initializeComponentsFactory.js` should we choose to do so.

Reply:
Making these dependencies explicit is great.  This highlights that initializeComponentsFactory is probably doing too much when we are passing in 9 dependencies.  This is beyond the scope of this PR, but I have some ideas on how to simplify this.